haystack/preview/document_stores/in_memory/document_store.py

```diff
@@ -92,75 +92,73 @@ def filter_documents(self, filters: Optional[Dict[str, Any]] = None) -> List[Doc
         """
         Returns the documents that match the filters provided.
 
-        Filters are defined as nested dictionaries. The keys of the dictionaries can be a logical operator (`"$and"`,
-        `"$or"`, `"$not"`), a comparison operator (`"$eq"`, `$ne`, `"$in"`, `$nin`, `"$gt"`, `"$gte"`, `"$lt"`,
-        `"$lte"`) or a metadata field name.
+        Filters are defined as nested dictionaries. There are two types of dictionaries:
+        - Comparison
+        - Logic
```

Comment:
I'm afraid docstrings are going to be unmaintainable... Do you know any way we can re-use the docstring from the protocol in all the different document stores?